bluesky_browser/suitcase.py

```diff
@@ -0,0 +1,389 @@
+    encoding = traitlets.Unicode('ascii')
+    compression = traitlets.Unicode('infer')
+    line_terminator = traitlets.Unicode('os.lineshape')
+    chunksize = traitlets.Int(None, allow_none=True)
+    date_format = traitlets.Unicode(None, allow_none=True)
+    doublequote = traitlets.Bool(True)
+    escapechar = traitlets.Unicode(None, allow_none=True)
+    decimal = traitlets.Unicode('.')
+
+    # Add the list of parameters for which the input widget should be displayed
+    parameters = traitlets.List(['directory', 'file_prefix', 'seperator'])
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def initGUI(self):
+        super().initGUI()  # ensures directory and file_prefix are at the top.
+
+        # set all of the unique traits widgets
+        self._create_trait_widget('seperator', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('na_rep', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('float_format', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('columns', QIntListEdit, 'setList',
+                                  'listChanged')
+        self._create_trait_widget('header', QStrListEdit, 'setList',
+                                  'listChanged')
+        self._create_trait_widget('index', QBoolBox, 'setBool',
+                                  'boolChanged')
+        self._create_trait_widget('index_label', QBoolNoneBox, 'setBool',
+                                  'boolChanged')
+        self._create_trait_widget('mode', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('encoding', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('compression', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('line_terminator', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('chunksize', QSpinBox, 'setValue',
+                                  'valueChanged')
+        self._create_trait_widget('date_format', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('doublequote', QBoolBox, 'setBool',
+                                  'boolChanged')
+        self._create_trait_widget('escapechar', QLineEdit, 'setText',
+                                  'textChanged')
+        self._create_trait_widget('decimal', QLineEdit, 'setText',
+                                  'textChanged')
+
+    def show_parameters(self, parameters=[]):
+        '''Walks through ``super().parameters``, ``self.parameters`` then
+        ``parameters`` showing the widgets in each.
+
+        NOTE: This modified version is required for all child classes. It
```

Comment:
It seemed to me there ought to be a better way than copying this into every child class. I pulled up [the traitlets API documentation](https://traitlets.readthedocs.io/en/stable/api.html#dynamic-default-values) and found that we are not the first people to need this kind of customization.

```python
In [16]: from traitlets import HasTraits, List, default

In [17]: class A(HasTraits):
    ...:     parameters = List()
    ...:     
    ...:     @default('parameters')
    ...:     def _get_default_parameters(self):
    ...:         return ['a', 'b']
    ...: 
    ...:         

In [18]: class B(A):
    ...:     @default('parameters')
    ...:     def _get_default_parameters(self):
    ...:         ret = super()._get_default_parameters()
    ...:         ret.extend(['c', 'd'])
    ...:         return ret
    ...:     

In [19]: a = A()

In [20]: a.parameters
Out[20]: ['a', 'b']

In [21]: b = B()

In [22]: b.parameters
Out[22]: ['a', 'b', 'c', 'd']
```